llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -428,16 +428,18 @@ Instruction *InstCombinerImpl::foldSelectOpOp(SelectInst &SI, Instruction *TI,
     CmpPredicate TPred, FPred;
     if (match(TI, m_ICmp(TPred, m_Value(), m_Value())) &&
         match(FI, m_ICmp(FPred, m_Value(), m_Value()))) {
-      // FIXME: Use CmpPredicate::getMatching here.
-      CmpInst::Predicate T = TPred, F = FPred;
-      if (T == F || T == ICmpInst::getSwappedCmpPredicate(F)) {
-        bool Swapped = T != F;
+      auto P = CmpPredicate::getMatching(TPred, FPred);
+      bool Swapped = !P;
+      if (Swapped)
+        P = CmpPredicate::getMatching(TPred,
+                                      ICmpInst::getSwappedCmpPredicate(FPred));
+      if (P) {
```

Comment:
This is pretty convoluted. I'd implement this by calling a lambda two times instead (once swapped).